plugins/src/webrtcsink/imp.rs

```diff
@@ -787,29 +779,132 @@ impl CongestionController {
         }
     }
 
-    fn clamp_bitrate(&mut self, bitrate: i32, n_encoders: i32) {
-        self.target_bitrate = bitrate.clamp(
-            self.min_bitrate as i32 * n_encoders,
-            self.max_bitrate as i32 * n_encoders,
+    fn clamp_bitrate(&mut self, bitrate: i32, n_encoders: i32, controller_type: ControllerType) {
+        match controller_type {
+            ControllerType::Loss => {
+                self.target_bitrate_on_loss = bitrate.clamp(
+                    self.min_bitrate as i32 * n_encoders,
+                    self.max_bitrate as i32 * n_encoders,
+                )
+            }
+
+            ControllerType::Delay => {
+                self.target_bitrate_on_delay = bitrate.clamp(
+                    self.min_bitrate as i32 * n_encoders,
+                    self.max_bitrate as i32 * n_encoders,
+                )
+            }
+        }
+    }
+
+    fn get_remote_inbound_stats(&self, stats: &gst::StructureRef) -> Vec<gst::Structure> {
+        let mut inbound_rtp_stats: Vec<gst::Structure> = Default::default();
+        for (_, field_value) in stats {
+            if let Ok(s) = field_value.get::<gst::Structure>() {
+                if let Ok(type_) = s.get::<gst_webrtc::WebRTCStatsType>("type") {
+                    if type_ == gst_webrtc::WebRTCStatsType::RemoteInboundRtp {
+                        inbound_rtp_stats.push(s);
+                    }
+                }
+            }
+        }
+
+        inbound_rtp_stats
+    }
+
+    fn lookup_rtt(&self, stats: &gst::StructureRef) -> f64 {
+        let inbound_rtp_stats = self.get_remote_inbound_stats(stats);
+        let mut rtt = 0.;
+        let mut n_rtts = 0u64;
+        for inbound_stat in inbound_rtp_stats.iter() {
```

Comment:
```suggestion
        for inbound_stat in &inbound_rtp_stats {
```

I see my suggestion got rid of the christmas tree below :)

Reply:
Yeah, thanks ;)